BOJ/11047.cpp

```diff
@@ -0,0 +1,32 @@
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+int main() {
+    cin.tie(nullptr)->ios_base::sync_with_stdio(false);
+
+    int N, K;
+    cin >> N >> K;
+
+    vector<int> coins;
+    int tmp;
+    while(N--){
+        cin >> tmp;
+        coins.push_back(tmp);
+    }
+
+    sort(coins.begin(), coins.end(), [](int a, int b){ return a > b; }); // 람다함수 이용 내림차순 정렬
+    // reverse(coins.begin(), coins.end()); // coins 배열이 오름차순임이 보장되므로 그냥 뒤집기. 어떤게 더 효과적..?
+
+    int cnt{};
+    auto iter = coins.begin();
+    while(K>0 && iter != coins.end()){
+        if(*iter > K){
+            iter++;
+            continue;
+        }; // 현재 단위가 남은 돈보다 크면 단위를 하나 낮추고 다시 반복
+        K -= *iter;
+        cnt++;
+    }
```

Comment:
피드백을 반영해 뺄셈을 나눗셈과 모듈로 연산으로 대체 >> 실행시간이 `60ms`에서 `0ms`로 감소함
```suggestion
    auto riter = coins.rbegin();
    while(K>0){ // 가치가 1인 동전이 있어서 항상 나누어 떨어지기 때문에 K값만 체크해도 문제없다
        if(*riter > K){
            riter++;
            continue;
        }; // 현재 단위가 남은 돈보다 크면 단위를 하나 낮추고 다시 반복
        cnt += K / *riter; // 현재 단위로 나눈 몫만큼 더하기. 좀 더 효율적인 코드!!
        K %= *riter; // 현재 단위로 나눈 나머지로 변경
    }
```